src/DGmethods/examples/ex_001_periodic_advection.jl

```diff
@@ -0,0 +1,438 @@
+  # we specify the following `Tuple` of `range`s:
+  brickrange = (range(DFloat(0); length=Ne+1, stop=1), # x_1 corner locations
+                range(DFloat(0); length=Ne+1, stop=1), # x_2 corner locations
+                range(DFloat(0); length=Ne+1, stop=1)) # x_3 corner locations
+  # these coordinates will be combined in a tensor product fashion to define the
+  # element corners.
+  #
+  # By default the `BrickTopology` is not periodic, so we need to define a
+  # `Tuple` of boolean defining which dimensions are periodic.
+  periodicity = (true, true, true)
+  # Note: We have defined both the `brickrange` and `periodicity` as though we
+  # are working in three-dimensions, and in when in two-dimensions we will
+  # discard the third element of the `Tuple`; this could also be done from the
+  # start by using the `ntuple` function.
+
+  # Using `brickrange` and `periodicity` we can now initialize the topology
+  # using the `BrickTopology` constructor. This will both create the topology as
+  # well as do the partitioning of the elements across the MPI ranks available
+  # in the mpi communicator `mpicomm`
+  topology = BrickTopology(mpicomm, brickrange[1:dim];
+                           periodicity=periodicity[1:dim])
+
+  # The topology only has element connectivity and corner information, thus we
+  # still need to create a grid (or mesh) of degrees of freedom. In CLIMA the
+  # so-called discontinuous spectral element grid is used (aka tensor-product
+  # quadrilateral and hexahedral elements with Legendre-Gauss-Lobatto
+  # interpolation and quadrature weights). Given a topology and polynomial order
+  # we can create the grid of degrees of free using
+  grid = DiscontinuousSpectralElementGrid(topology; polynomialorder =
+                                          polynomialorder, FloatType = DFloat,
+  DeviceArray = ArrayType,)
+  # Note: This constructor also takes in a `FloatType` which specifies the
+  # floating point type to be used (e.g., for the coordinate points and geometry
+  # metric terms). The argument `ArrayType` is used to determine the compute
+  # device to use (i.e., `Array` will signify the CPU is being used and
+  # `CuArray` will signal that an NVIDIA GPU is being used).
+
+  # We can now define the discretization from the `grid` using the physical flux
+  # `advectionflux!` and numerical flux `upwindflux!` defined above; we only
+  # have a single state variable $q$ hence `length_state_vector = 1`
+  spatialdiscretization = DGBalanceLaw(grid = grid, length_state_vector = 1,
+                                       inviscid_flux! = advectionflux!,
+                                       inviscid_numerical_flux! = upwindflux!)
+
+  # (end of function)
+end
+#md nothing # hide
+
+#------------------------------------------------------------------------------
+
+# ### Initializing and run the DG method
+# Note: This whole code chunk is in a `let` block
+let
```

Comment:
Out of curiosity, why run the DG method in a `let` block?

Reply:
I just like to protect my repl-scope

Reply:
Not sure if there is a performance impact too with function closure?